### PR TITLE
Ruelvls 24 refact

### PR DESCRIPTION
**Pull request by DianaBernardo**
Created Archetype and Archetype DTO with @BPAndrea's help. Can you please have a final look?

### CODE REVIEW

File: src/main/java/com/greenfox/lvlup/LvlupApplication.java [hunk 1]
```diff
@@ -1,14 +1,26 @@
...
+import com.greenfox.lvlup.repositrory.BadgeLevelRepository;
+import com.greenfox.lvlup.repositrory.BadgeRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.CommandLineRunner;
 import org.springframework.boot.SpringApplication;
 import org.springframework.boot.autoconfigure.SpringBootApplication;
 
 @SpringBootApplication
-public class LvlupApplication {
+public class LvlupApplication implements CommandLineRunner {
```
Comment (drexeverest) [inline src/main/java/com/greenfox/lvlup/LvlupApplication.java]: Atm CommandLineRunner is not used, and we should remove unused code parts

File: src/main/java/com/greenfox/lvlup/controller/ArchetypeController.java [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  ArchetypeRepository archetypeRepository;
+  ArchetypeService archetypeService;
+
+  @Autowired
+  public ArchetypeController(ArchetypeRepository archetypeRepository, ArchetypeService archetypeService) {
+    this.archetypeRepository = archetypeRepository;
+    this.archetypeService = archetypeService;
+  }
+
+  /*
```
Comment (drexeverest) [inline src/main/java/com/greenfox/lvlup/controller/ArchetypeController.java]: Remove unnecessary comments/codes plz :)

File: src/main/java/com/greenfox/lvlup/controller/ArchetypeController.java [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+  @GetMapping("/archetypedto")
+  public ArchetypeDTO getArchetypeDTO() {
+    return archetypeService.convertArchetypeToDTO(archetypeService.generateArchetype());
+  }*/
+
+ @GetMapping("/archetypes")
```
Comment (drexeverest) [inline src/main/java/com/greenfox/lvlup/controller/ArchetypeController.java:25]: Remove if it's not necessary, but do not use JsonIgnore

File: src/main/java/com/greenfox/lvlup/model/entity/Archetype.java [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+
+  public String getName() {
+    return name;
+  }
+
+  public void setName(String name) {
+    this.name = name;
+  }
+
+
```
Comment (drexeverest) [inline src/main/java/com/greenfox/lvlup/model/entity/Archetype.java]: ➖line

**Review by drexeverest — CHANGES_REQUESTED** [review]

**Comment by DianaBernardo** [comment]
I refactored the code based on the code review comments.

**Review by drexeverest — APPROVED** [review]

**Pull request merged**